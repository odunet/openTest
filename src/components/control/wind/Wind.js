import React, { useState, useContext } from 'react';
import { TextLabel, FlexColDiv, Radio } from '../../../shared/styledComponents';
import { displayWind_Action } from '../../../actions';
import { store } from '../../../store';
import { handleCheckWind } from '../../../helper/handleCheck';

export default function Wind() {
  const [radio, setRadio] = useState('on');

  // Bring in dispatch
  const { dispatch } = useContext(store);

  return (
    <>
      <TextLabel htmlFor='controlTitle'>Wind</TextLabel>
      <FlexColDiv>
        <FlexColDiv>
          <Radio
            type='radio'
            id='on'
            name='wind'
            value='on'
            onChange={(e) =>
              handleCheckWind(dispatch, setRadio, displayWind_Action, e.target)
            }
            checked={radio === 'on'}
          />{' '}
          <p>On</p>
        </FlexColDiv>
        <FlexColDiv>
          <Radio
            type='radio'
            id='off'
            name='wind'
            value='off'
            onChange={(e) =>
              handleCheckWind(dispatch, setRadio, displayWind_Action, e.target)
            }
            checked={radio === 'off'}
          />
          <p>Off</p>
        </FlexColDiv>
      </FlexColDiv>
    </>
  );
}
