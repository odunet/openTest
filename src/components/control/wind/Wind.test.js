import { render, screen } from '@testing-library/react';
import Wind from './Wind';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, mount } from 'enzyme';

describe('Render Temperature UI', () => {
  test('renders Wind', () => {
    render(<Wind />);
    const windElement = screen.getByText(/Wind/i);
    expect(windElement).toBeInTheDocument();
  });
});

describe('Snapshot', () => {
  test('Should take a snapshot of Wind component', () => {
    const container = shallow(<Wind />);
    expect(toJson(container)).toMatchSnapshot();
  });
});
