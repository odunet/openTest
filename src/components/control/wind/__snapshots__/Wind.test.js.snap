// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should take a snapshot of Wind component 1`] = `
<Fragment>
  <styled.label
    htmlFor="controlTitle"
  >
    Wind
  </styled.label>
  <styled.div>
    <styled.div>
      <styled.input
        checked={true}
        id="on"
        name="wind"
        onChange={[Function]}
        type="radio"
        value="on"
      />
       
      <p>
        On
      </p>
    </styled.div>
    <styled.div>
      <styled.input
        checked={false}
        id="off"
        name="wind"
        onChange={[Function]}
        type="radio"
        value="off"
      />
      <p>
        Off
      </p>
    </styled.div>
  </styled.div>
</Fragment>
`;
